internal/controller/discoverymgr/discovery.go

```diff
@@ -251,8 +252,8 @@ func (d *DiscoveryImpl) ResetServiceName() {
 // AddDeviceInfo takes public and private IP from relay and requests for orchestration info from this device
 func (d *DiscoveryImpl) AddDeviceInfo(deviceID string, virtualAddr string, privateAddr string) {
 
-	log.Println(logPrefix, "[AddDeviceInfo]", "private Addr", privateAddr)
-	log.Println(logPrefix, "[AddDeviceInfo]", "Virtual Addr", virtualAddr)
+	log.Println(logPrefix, "[AddDeviceInfo]", "private Addr"+strings.ReplaceAll(fmt.Sprintf("%#v", privateAddr), "\"", "")) // lgtm [go/log-injection]
```

Comment:
I found recommendation of code scanning alert changed \n and \r to blank. You change only " to blank. Is it enough?

Reply:
About replacing `\n` `\r` characters with see above. Also yesterday I sent you another version of the function:
```
func SanitizeUserInput(input string) string {
    output := strings.ReplaceAll(fmt.Sprintf("%#v",input),"\"","")
    //output := strings.ReplaceAll(strings.ReplaceAll(input,"\n",""),"\r","")
    if !strings.Compare(input, output) {
        logIns.Warn("Possibly the following record contains a malicious injection")
    }
    return output
}
```

**Below two options:**
_I liked the first one because analyzing the log it is possible to determine an attempted attack, and the input parameter can be of any type_ 
```
package main

import (
	"fmt"
	"strings"
)

func main() {
	var input =  "This is a original string for test!\n\rFake (malicious) sub-string for test!"

	fmt.Println("Print without sanitize:", input)
	fmt.Println("")

	fmt.Println("Using: strings.ReplaceAll(fmt.Sprintf(\"%#v\", input), \"\"\",\"\")")
	fmt.Println("Test:",strings.ReplaceAll(fmt.Sprintf("%#v", input), "\"",""))
	fmt.Println("")

	fmt.Println("Using: strings.ReplaceAll(fmt.Sprintf(\"%#v\", input), \"\"\",\"\")")
	fmt.Println("Test:",strings.ReplaceAll(strings.ReplaceAll(input,"\n",""),"\r",""))
	fmt.Println("")
}
```
Result:
```
Print without sanitize: This is a original string for test!
Fake (malicious) sub-string for test!

Using: strings.ReplaceAll(fmt.Sprintf("%#v", input), ""","")
Test: This is a original string for test!\n\rFake (malicious) sub-string for test!

Using: strings.ReplaceAll(fmt.Sprintf("%#v", input), ""","")
Test: This is a original string for test!Fake (malicious) sub-string for test!
```

Reply:
About replacing `\n` `\r` characters with see above. Also yesterday I sent you another version of the function:

```
func SanitizeUserInput(input string) string {
    output := strings.ReplaceAll(fmt.Sprintf("%#v",input),"\"","")
    //output := strings.ReplaceAll(strings.ReplaceAll(input,"\n",""),"\r","")
    if !strings.Compare(input, output) {
        logIns.Warn("Possibly the following record contains a malicious injection")
    }
    return output
}
```
**Below two options:**
_I liked the first one because analyzing the log it is possible to determine an attempted attack, and the input parameter can be of any type_

```
package main

import (
	"fmt"
	"strings"
)

func main() {
	var input =  "This is a original string for test!\n\rFake (malicious) sub-string for test!"

	fmt.Println("Print without sanitize:", input)
	fmt.Println("")

	fmt.Println("Using: strings.ReplaceAll(strings.ReplaceAll(input,\"\\n\",\"\"),\"\\r\",\"\")")
	fmt.Println("Test:",strings.ReplaceAll(fmt.Sprintf("%#v", input), "\"",""))
	fmt.Println("")

	fmt.Println("Using: strings.ReplaceAll(fmt.Sprintf(\"%#v\", input), \"\"\",\"\")")
	fmt.Println("Test:",strings.ReplaceAll(strings.ReplaceAll(input,"\n",""),"\r",""))
	fmt.Println("")
}
```
Result:
```
Print without sanitize: This is a original string for test!
Fake (malicious) sub-string for test!

Using: strings.ReplaceAll(fmt.Sprintf("%#v", input), ""","")
Test: This is a original string for test!\n\rFake (malicious) sub-string for test!

Using: strings.ReplaceAll(strings.ReplaceAll(input,"\n",""),"\r","")
Test: This is a original string for test!Fake (malicious) sub-string for test!

```

Reply:
The first approach looks good to me. Rather than sanitize each log individually, it would be good to have one common function.
Can you implement sanitize function under `internal/common/logmgr/` or `securemgr`?

Current:
log.Println(logPrefix, "[AddDeviceInfo]", "private Addr"+strings.ReplaceAll(

To be:
log.Println(logPrefix, "[AddDeviceInfo]", "private Addr"+log.SanitizeUserInput(

Reply:
log.SanitizeUserInput(privateAddr) => logmgr.SanitizeUserInput(privateAddr)